src/software/ai/navigator/path_planner/theta_star_path_planner_test.cpp

```diff
@@ -71,26 +71,31 @@ TEST(TestThetaStarPathPlanner, test_theta_star_path_planner_blocked_src)
     std::vector<Obstacle> obstacles = {Obstacle(
         Polygon({Point(0.5, 1), Point(-0.5, 1), Point(-0.5, -1), Point(0.5, -1)}))};
 
-    std::unique_ptr<PathPlanner> planner =
-        std::make_unique<ThetaStarPathPlanner>(field, obstacles);
+    std::unique_ptr<PathPlanner> planner = std::make_unique<ThetaStarPathPlanner>();
+    std::vector<Point> path_points;
 
-
-    auto path_points = planner->findPath(start, dest);
-
-    // We expect to find a path
-    ASSERT_TRUE(path_points);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
+            planner->findPath(start, dest, field, obstacles));
+    }
+    catch (const std::bad_variant_access&)
+    {
+        // not a valid path
+        EXPECT_TRUE(0);
```

Comment:
gtest actually has a nice mechanism to explicitly fail, see https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#explicit-success-and-failure

Reply:
ok

Reply:
Done